### PR TITLE
Shell command

### PR DESCRIPTION
**Pull request by echo-bravo-yahoo**
Ticket(s): FE-6124

Implements the `shell` functionality of the old CLI.

Additionally:
- adds stdout/stderr to the DI container to allow tests to assert on the whole output.

### CODE REVIEW

File: test/helpers.mjs [hunk 1]
```diff
@@ -29,3 +32,63 @@ export function buildUrl(path, paramObject) {
...
+      console.log(`Byte ${i}: ${charCode1}, ${charCode2}`);
+  }
+}
+
+/**
+ * Class representing a no-frills writable stream that, instead of writing data
+ * to a destination, holds it in memory. Can be queried for this data later;
+ * use it for testing interfaces that use streams.
+ */
+export class InMemoryWritableStream extends Writable {
```
Comment (echo-bravo-yahoo) [inline test/helpers.mjs:70]: Test writable stream. For readable streams, I've just been using `Readable.from()` and then using the native methods on it.

File: test/login.mjs [hunk 2]
```diff
@@ -90,16 +92,14 @@ describe("login", function () {
     expect(oauthClient.start.called).to.be.true;
     // We open auth url in the browser and prompt user
     expect(container.resolve("open").calledWith("dashboard-url"));
-    expect(
-      logger.stdout.calledWith(
```
Comment (echo-bravo-yahoo) [inline test/login.mjs:94]: This syntax actually can't fail - gotta move the calledWith outside the expect.

File: src/config/setup-test-container.mjs [hunk 3]
```diff
@@ -49,24 +55,7 @@ export function setupTestContainer() {
...
-      error: logger.error,
-      fatal: logger.fatal,
-
-      // use these for communicating with customers.
-      // mocked because they _are_ part of our public
-      // interface and should be tested.
-      stdout: stub(),
-      stderr: stub(),
-    }),
+    logger: awilix.asFunction((cradle) => spy(buildLogger(cradle))).singleton(),
```
Comment (echo-bravo-yahoo) [inline src/config/setup-test-container.mjs:58]: I really like this pattern - spy on building a singleton injectable.

File: src/commands/shell.mjs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    preview: argv.version !== "10",
+    // TODO: integrate with fql-analyzer for completions
+    completer: argv.version === "10" ? () => [] : undefined,
+    output: container.resolve("stdoutStream"),
+    input: container.resolve("stdinStream"),
+    eval: await buildCustomEval(argv),
+    terminal: true,
+  };
+
+  const shell = repl.start(replArgs);
```
Comment (echo-bravo-yahoo) [inline src/commands/shell.mjs:32]: like the prior CLI, this uses node's REPL, which gives us lots of thing for free (syntax highlighting in some cases, standard keyboard shortcuts, prompts, etc).

**Review by mwilde345 — APPROVED** [review]

**Pull request merged**